src/experimentation-ess/common.js

```diff
@@ -418,6 +418,45 @@ function getObjectByProperty(array, name, value) {
   return array.find((e) => e[name] === value);
 }
 
+/**
+ * returns the date n days ago
+ * @param {number} days
+ * @returns {Date} the date n days ago
+ */
+function getDate(days) {
+  const date = new Date();
+  date.setDate(date.getDate() - days);
+  date.setHours(0, 0, 0, 0);
+  return date;
+}
+
+/**
+ * Updates the experiment start, end dates and status based on the inferred dates
+ * @param {*} experiments
+ */
+function updateExperimentMetadata(experiments) {
+  // get the date 2 days ago to account for timezones
+  const yesterday = getDate(2);
+  for (const experiment of experiments) {
+    experiment.startDate = experiment.startDate || experiment.inferredStartDate;
+    const inferredEndDate = new Date(experiment.inferredEndDate);
+    inferredEndDate.setHours(0, 0, 0, 0);
+    if (inferredEndDate >= yesterday) {
+      // found RUM data yesterday, so endDate should be either in the metadata or null
+      experiment.endDate = experiment.endDate || null;
+    } else {
+      experiment.endDate = experiment.endDate || experiment.inferredEndDate;
+    }
+    if (!experiment.status) {
+      if (experiment.endDate && new Date(experiment.endDate) < new Date()) {
+        experiment.status = 'COMPLETE';
```

Comment:
*nitpicking*: I'm not sure I'd chose "complete" as a status. To me, it kinda means we have reached completion of the experiment which would mean both that it ended and that we reached stats sig. The latter is not guaranteed here though. So from a pure semantics point-of-view, I'd be tempted to go with "inactive"/"stopped".
"Inactive" is what we usually have in the metadata and documented, so maybe that would be the most appropriate?